ludwig/utils/trainer_utils.py

```diff
@@ -408,3 +410,20 @@ def get_rendered_batch_size_grad_accum(config: "BaseTrainerConfig", num_workers:
                 gradient_accumulation_steps = 1
 
     return batch_size, gradient_accumulation_steps
+
+
+def freeze_layers_regex(config: "BaseTrainerConfig", model: ECD) -> None:
+    """Freezes layers based on provided regular expression."""
+    try:
+        pattern = re.compile(config.layers_to_freeze_regex)
```

Comment:
@ethanreidel Would you be interested if I gave you a reasonably well-featured RegEx utility so that you can just put it into the utils and use it -- it will save a lot of boilerplate like this.  Please let me know any time.  Thanks!

Reply:
Yeah that sounds good. Thanks